docs: Add function call example for no-undefined (#16712)
eslint/eslint: docs/src/rules/no-undefined.md

function foo(undefined) {
    // ...
}

// Function creation with parameter
function bar(baz, qux) {
    // ...
}

> // Using function with improper refrence to the global undefined value to attempt to skip the first parameter

Reviewer: ```suggestion
// Using a function with improper reference to the undefined global value to attempt to skip the first parameter
```

> // Using function with improper refrence to the global undefined value to attempt to skip the first parameter

Author: I agree

function foo(undefined) {
    // ...
}

> // Skips the first parameter

Reviewer 2: This is a bit confusing, what do you mean by "skips"? I think this comment is making it more confusing and we can just remove it.

This is in the "incorrect" section, so this would be flagged as incorrect by `no-undefined`

Author: Would the original comments (plus spelling fixes) be better?

Reviewer 3: I think it would be best to just remove the comments as the intent seems clear from the code.

}

global.undefined = "foo";

> // Skips the first parameter

Reviewer 2: Same comment here. I'm not sure what this is supposed to mean.

Author: Would the original comments (plus spelling fixes) be better?

}

global.undefined = "foo";

// Function creation with parameter
function bar(baz, qux) {
    // ...
}

> // Using function wihtout refrencing the global undefined

Reviewer: ```suggestion
// Using function without referencing the global undefined
```

> // Function creation with parameter
> function bar(baz, qux) {
>     // ...
> }
> 
> // Using function wihtout refrencing the global undefined

Reviewer 3: ```suggestion
// Skips the first parameter
```

Same here.

Author: I agree
